data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,247 @@
+}
+set output \$expect_out(buffer)
+expect eof
+
+puts \"\$output\"
+"
+
+}
+
+function open_console() {
+
+    #open guest console
+    for vmguest in ${vm_guestnames_state_running_array[@]};do
+        get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+    done 
+}
+
+function close_console() {
+
+    #close guest console
+    echo -e "Kill PIDs $(basename $0)\n" >> ${monitor_log_file}
+    pid=`ps aux | grep "$(basename $0) -s" | grep -v "grep"| awk '{print $2}'`
+    if [[ -n $pid ]];then
+        kill -9 $pid >> /dev/null 2>&1;sync
+    else
+        echo -e "PIDs $(basename $0) were killed\n" >> ${monitor_log_file}
+    fi
+
+    echo -e "Kill all associated "virsh console" processes\n" >> ${monitor_log_file}
+
+    while :
+    do
+        #wait for checking all associated "virsh console" processes"
+        sleep 3
+
+        #Check all associated "virsh console" processes
+        ps aux | grep "virsh console" | grep -v "grep" >> /dev/null 2>&1
+        if [[ $? -eq 0 ]];then 
+            sub_pid=`ps aux | grep "expect -c" | grep -v "grep"| awk '{print $2}'`
+            if [[ -n $sub_pid ]];then
+                kill -9 $sub_pid >> /dev/null 2>&1;sync
+            fi
+        else
+            break
+        fi
+    done
+}
+
+function end_monitor() {
+
+    #end of monitor for guest console
+    if [ -f ${MONITOR_LOCKED_FILE} ];then
+        echo -e "End of Monitor for guest console:\n" | tee -a ${monitor_log_file}
+        #UNLOCK Monitor for guest console
+        rm -rf ${MONITOR_LOCKED_FILE} >> /dev/null 2>&1
+        #Kill Pids from Monitor for guest console
+        close_console
+    fi
+}
+
+function start_monitor() {
+
+    #Start monitor for guest console
+    ##After boot up all existed vm guest(s) from set_guest_running func
+    ##Figure out vm_guestnames_state_running_array from all existed and boot up vm guest(s) 
+    get_vm_guestnames_state_running=`virsh list --state-running | grep "${vm_guestnames_types}" | awk '{print $2}'`
+    vm_guestnames_state_running_array=$(echo -e ${get_vm_guestnames_state_running})
+
+    while :
+    do
+        if [ -f ${MONITOR_LOCKED_FILE} ];then
+            #Check with LOCK-UP status
+            cat ${MONITOR_LOCKED_FILE} | grep "LOCKED" >> /dev/null 2>&1
+            if [ $? -eq 0 ];then
+                #KEEP LOCK-UP status
+                #To let just only one virsh console with one vm guest as backgroup PID
+
+                #wait for checking background PID of virsh console
+                sleep 3
+		#Check with background PID of virsh console
+		for vmguest in ${vm_guestnames_array[@]};do
+		    virsh list --state-running | grep ${vmguest}
+                    if [[ $? -eq 0 ]];then
+                        ps aux | grep "virsh console ${vmguest}" | grep -v grep
+                        if [[ $? -ne 0 ]];then
+                            get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
```

Comment:
THis does not seems a good idea. When people want to start monitor by "this script -s", they should only need serial log from this point on, but they do not need earlier logs. However this solution will deliver earlier logs. Besides for multiple guest scenarios, especially when some guest has console open while some not, it will upload logs starting from different time.

Suggest to take an easier solution. When people want to start console by "[this script] -s ", firstly cleanup env by killing all console related processes including "virsh console"/expect by end_monitor, then start console freshly .

Reply:
Cool, this is a great idea.
Refer to the current solution, user can use this script anytime to enable|disable handle guest serial console as required, and then refer to guest_console log by timestamp

Also, will add step - firstly cleanup env by killing all console related processes including "virsh console"/expect by end_monitor, then start console freshly before start console by "[this script] -s ". Pretty cool!

Reply:
If you plan to adopt this easier solution, it seems that the LOCK file is not needed any more, because you will forever try to end console first no matter virsh console is started or not, no matter lock file exists or not. Right?

Reply:
no, I do not agree this. 
As mentioned, Start Monitor for guest console
Collect the existed vm guest(s) on vm host -> try to boot up the existed vm guest(s).
**-> Create LOCK-UP file to keep virsh console PID(s) still running as background**
-> call get_console() to made virsh console as child process, which was ran as background PID and redirect
output of virsh console to specified log file

instance :
If this LOCK-UP file was existed mean that virsh console process ran as background very well without any effected from outside as the benchmark, follow this rule, if the existed vm guest keep running status, but figure out virsh console process was gone, then re-trigger this virsh console process again - handle guest console process keel still running as purpose

Reply:
I know this is your first design's idea. Suggest you look again at your previous reply to recall what you plan to change and why you agree that. Previous comment link: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10797#discussion_r469692908

Reply:
sure, agree to changed as below:
1. each guest console just have only one console log  - DONE  
Such as:
get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console.log
2. firstly cleanup env by killing all console related processes including "virsh console"/expect by end_monitor, then start console freshly before start console by "[this script] -s "

Reply:
firstly cleanup env by killing all console related processes including "virsh console"/expect by end_monitor, then start console freshly before start console by "[this script] -s " - DONE

Reply:
DONE